EIPS/eip-5299.md

```diff
@@ -0,0 +1,165 @@
+```
+// A contract that implements Parent Storage and Implementation contract storage.
+library LibB {
+
+    // This struct contains state variables related to diamond
+    struct ParentStorage {
+        address owner;
+    }
+
+    // This struct contains state variables related to specific 
+    struct AppStorageV1 {
+        bytes32 dataA;
+    }
+
+    // Upgraded struct
+    struct AppStorageV2 {
+        uint256 dataA;
+    }
+
+    // Returns the struct from a specified position in contract storage
+    // ps is short for DiamondStorage
+    function parentStorage() internal pure returns(ParentStorage storage ps) {
+        bytes32 storagePosition = keccak256("parent.storage");
+        // Set the position of our struct in contract storage
+        assembly {ps.slot := storagePosition}
+    }
+
+    // Returns the struct from a specified position in contract storage
+    // as is short for AppStorage
+    function appStorageV1() internal pure returns(AppStorageV1 storage as) {
+        bytes32 storagePosition = keccak256("app.storage.implContractA.V1");
+        assembly {as.slot := storagePosition}
+    }
+
+    // Returns the struct from a specified position in contract storage
+    // as is short for AppStorage
+    function appStorageV2() internal pure returns(AppStorageV2 storage as) {
+        bytes32 storagePosition = keccak256("app.storage.implContractA.V2");
+        assembly {as.slot := storagePosition}
+    }
+}
+
+// Our implementation contract uses the Parent Storage defined above along with the implementation contract specific storage
+contract ImplContractAV2 {
+    function setDataAV2(uint256 _dataA) external {
+        LibB.DiamondStorage storage ds = LibA.parentStorage();
+        LibB.AppStorageV2 storage as = LibA.appStorageV2();
+        require(ds.owner == msg.sender, "Must be owner.");
+        as.dataA = _dataA;
+  }
+
+    function getDataAV2() external view returns (uint256) {
+        LibB.AppStorageV2 storage as = LibA.appStorageV2();
+        return as.dataA;
+    }
+
+    function getDataAV1AndV2() external view returns (bytes32, uint256) {
+        LibB.AppStorageV1 storage as1 = LibA.appStorageV1();
+        LibB.AppStorageV2 storage as2 = LibA.appStorageV2();
+        return (as1.dataA, as2.dataA);
+    }
+}
+
+```
+With above implementation, we will have access to both AppStorageV1 and AppStorageV2 which can be used either for migration or used in conjunction as per the use case.
+
+## Rationale
+In order to make the upgrade process as smooth and seamless possible it is mandatory to keep all the components involved in a system as modular as possible. Hence the upgradable storage structure will prove to be very helpful.
```

Comment:
```suggestion
WIP
```

A rationale explains design decisions made in the Specification (e.g. why a function is named like it is, why a specific type is used instead of another, etc...)

While it is a required section, placeholders for this specific section are common enough that I would no longer consider having a full Rationale necessary for an EIP to be merged as Draft.